core/src/err.rs

```diff
@@ -23,6 +35,9 @@ impl Error {
                 GENERAL_ERROR
             }
             Error::MongoParseConnectionStringError(_) => UNABLE_TO_CONNECT,
+            Error::NoDatabase => NO_DSN_OR_DRIVER,
+            Error::ColIndexOutOfBounds(_) => COLUMN_NOT_FOUND,
+            Error::BsonDeserialization(_) | Error::ValueAccess(_)  | Error::InvalidResultSetJsonSchema => GENERAL_ERROR, // TODO: might want to do invalid cursor state?
```

Comment:
I'm not sure what ODBC error code we want to map some of these internal errors to. They don't have clear corresponding error codes as far as I can tell. This is likely because the ODBC spec doesn't anticipate this kind of work: retrieving schema information from some other source and deserializing/manipulating it in error-prone ways. I went with GENERAL_ERROR, but INVALID_CURSOR_STATE could be a more specific alternative...maybe?

Reply:
I'd move the schema retrieval before the query execution and return GENERAL_ERROR. The less we have to worry about cursors, the better it will be :)